src/journal/journal.c

```diff
@@ -393,9 +408,42 @@ static int is_class_prefix(char *class)
         return (strcmp((char *)(class + class_len - 2), "/*") == 0) ? 1 : 0;
 }
 
+/**
+ * Print records content
+ *
+ * @param record_id Unique record identifier
+ *
+ */
+static void print_record(char *record_id)
+{
+        int rc = 0;
+        size_t read = 0;
+        char buff[MID_BUFF] = { '\0' };
+        char *filepath = NULL;
+        FILE *recordfp = NULL;
+
+        rc = asprintf(&filepath, "%s/%s", RECORD_RETENTION_DIR, record_id);
+        if (rc == -1) {
+                // just bail out, there are worse problems
+                return;
+        }
+
+        recordfp = fopen(filepath, "r");
+        if (!recordfp) {
+                return;
+        }
+
+        while ((read = fread(buff, 1, sizeof(buff), recordfp)) > 0) {
+                puts(buff);
+        }
+
+        free(filepath);
+        fclose(recordfp);
+}
+
 /* Exported function */
 int print_journal(TelemJournal *telem_journal, char *classification,
-                  char *record_id, char *event_id, char *boot_id)
+                  char *record_id, char *event_id, char *boot_id, int record)
```

Comment:
Does record only get used as a boolean at line 510 below? Might just make it a bool to make it clear.